gcore/geoheif.h

```diff
@@ -0,0 +1,53 @@
+/******************************************************************************
+ * Project:  GeoHEIF support class
+ * SPDX-License-Identifier: MIT
+ ****************************************************************************/
+
+#ifndef GEOHEIF_H_INCLUDED_
+#define GEOHEIF_H_INCLUDED_
+
+#include "gdal_pam.h"
+#include "ogr_spatialref.h"
+
+#include <vector>
+
+//! @cond Doxygen_Suppress
+
+/*
+ * GeoHEIF support implementation.
+ * 
+ * This class provides shared implementation for OGC GeoHEIF georeferencing,
+ * which is currently in draft (see OGC 24-038).
+ * 
+ * GeoHEIF provides parsing and caching for spatial references, pixel
+ * to model affine transformation, and tie-points.
+ * 
+ * This class is only shared here to provide common usage within
+ * AVIF and HEIF drivers. It is not intended to be a user-level API.
+*/
+class CPL_DLL GeoHEIF final
```

Comment:
either rename the class a GDALGeoHEIF, or put it withing a "gdal" namespace

Reply:
Moved it into the "gdal" namespace.